src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +387,160 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
                 }
 
-                manifest.AddLibrary(new LibraryInstallationState
+                LibraryInstallationState libraryInstallationState = new LibraryInstallationState
                 {
                     LibraryId = PackageId,
                     ProviderId = selectedPackage.ProviderId,
                     DestinationPath = InstallationFolder.DestinationFolder,
                     Files = SelectedFiles.ToList()
-                });
+                };
 
-                await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                await Shell.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
                 EnvDTE.Project project = VsHelpers.DTE.SelectedItems.Item(1)?.ProjectItem?.ContainingProject;
                 project?.AddFileToProjectAsync(_configFileName);
 
-                await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                RunningDocumentTable rdt = new RunningDocumentTable(Shell.ServiceProvider.GlobalProvider);
+                string configFilePath = Path.GetFullPath(_configFileName);
 
-                _dispatcher.Invoke(() =>
+                IVsTextBuffer document = rdt.FindDocument(configFilePath) as IVsTextBuffer;
+
+                // The file isn't open. So we'll write to disk directly
+                if (document == null)
                 {
-                    _closeDialog(true);
-                });
+                    manifest.AddLibrary(libraryInstallationState);
+
+                    await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+
+                    await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
+                }
+                else
+                {
+                    // libman.json file is open, so we will write to the textBuffer.
+                    _dispatcher.Invoke(() =>
+                    {
+                        _closeDialog(true);
+                    });
+
+                    InsertIntoTextBuffer(document, libraryInstallationState);
+
+                    // Save manifest file so we can restore library files.
+                    rdt.SaveFileIfDirty(configFilePath);
+                }
             }
             catch { }
         }
 
+        private void InsertIntoTextBuffer(IVsTextBuffer document, LibraryInstallationState libraryInstallationState)
+        {
+            ITextBuffer textBuffer = GetTextBuffer(document);
+
+            if (textBuffer != null)
+            {
+                string insertionText;
+
+                JSONMember libraries = GetLibraries(textBuffer);
+
+                if (libraries != null && libraries.Children != null)
```

Comment:
>libraries != null && libraries.Children != null [](start = 20, length = 47)

nit: 

Could be 

if (libraries?.Children != null)

and insertionText declaration could be moved inside

Reply:
Updated in version 10.

---
In reply to: [197933718](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197933718) [](ancestors = 197933718)